### PR TITLE
PMM-5980 Add suspend pxc.

### PR DESCRIPTION
**Pull request by askomorokhov**
https://jira.percona.com/browse/PMM-5980

https://github.com/percona-platform/dbaas-controller/pull/102
https://github.com/percona-platform/dbaas-api/pull/70/files

### CODE REVIEW

File: api/managementpb/dbaas/xtradb_cluster.proto [hunk 1]
```diff
@@ -103,6 +103,8 @@ message ListXtraDBClustersResponse {
     RunningOperation operation = 3;
     // Cluster parameters.
     XtraDBClusterParams params = 4;
+    // Paused is true when cluster is suspended.
+    bool paused = 5;
```
Comment (BupycHuk) [inline api/managementpb/dbaas/xtradb_cluster.proto:107]: IMHO, it's better to handle this on backend side and just return new XtraDBClusterState, WDYT?

File: api/managementpb/dbaas/xtradb_cluster.proto [hunk 2]
```diff
@@ -197,12 +199,10 @@ message UpdateXtraDBClusterRequest {
...
     // Cluster size.
-    int32 cluster_size = 1 [
-      (validator.field) = {
-        int_gt: 0
-      }
-    ];
+    int32 cluster_size = 1;
     // PXC container parameters.
     PXC pxc = 2 [
       (validator.field) = {
```
Comment (BupycHuk) [inline api/managementpb/dbaas/xtradb_cluster.proto:208]: I think we should make these fields optional too.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: api/managementpb/dbaas/xtradb_cluster.proto (hunk 1 above)
Comment (percona-csalguero) [reply api/managementpb/dbaas/xtradb_cluster.proto:107]: :+1:

**Review by percona-csalguero — COMMENTED** [review]
I agree with Nurlan. Paused can be a new state

File: api/managementpb/dbaas/xtradb_cluster.proto (hunk 1 above)
Comment (AlekSi) [reply api/managementpb/dbaas/xtradb_cluster.proto:107]: Is it even a separate state on the operator level? Or that's just scaling to zero?

Comment (askomorokhov) [reply api/managementpb/dbaas/xtradb_cluster.proto:107]: Yes, we can do this to make UI more simple.
But we go away from what we really have in k8s and mix statuses with the other conception.

@AlekSi, no it is not separate state on operator level, this is scaling it to zero.
```
als@mbp percona-xtradb-cluster-operator % kubectl get pxc/my-test-pxc -o json | jq .status.state
"ready"
als@mbp percona-xtradb-cluster-operator % kubectl get pod                                       
NAME                                               READY   STATUS    RESTARTS   AGE
percona-xtradb-cluster-operator-5c99579f95-k82p4   1/1     Running   0          16h
als@mbp percona-xtradb-cluster-operator % kubectl get pxc/my-test-pxc -o json | jq .spec.pause  
true
als@mbp percona-xtradb-cluster-operator % kubectl get pvc | grep my-test-pxc                     
datadir-my-test-pxc-pxc-0   Bound    pvc-11a1e4be-4f71-458a-914d-32f5da1e9431   6Gi        RWO            gp2            29m
datadir-my-test-pxc-pxc-1   Bound    pvc-40efe271-814d-4edc-bd03-74c38dfbb0a0   6Gi        RWO            gp2            28m
datadir-my-test-pxc-pxc-2   Bound    pvc-345fd6d1-0f39-44b4-8f22-01c06934bfe3   6Gi        RWO            gp2            27m
```

Comment (BupycHuk) [reply api/managementpb/dbaas/xtradb_cluster.proto:107]: We use separate cluster states in our API, so I think we can add one more state to our states list.

File: api/Makefile [hunk 3]
```diff
@@ -39,8 +39,10 @@ gen: clean            ## Generate files.
 			--additional-initialism=pt \
 			--additional-initialism=qan \
 			--additional-initialism=rds \
-			--additional-initialism=sql ; \
+			--additional-initialism=psmdb \
+			--additional-initialism=sql & ); \
 	done
+	wait
```
Comment (askomorokhov) [inline api/Makefile:45]: #sequentially:
make -C api gen  48.70s user 13.75s system 225% cpu 27.731 total
make -C api gen  50.94s user 14.08s system 214% cpu 30.354 total
make -C api gen  48.89s user 13.23s system 226% cpu 27.392 total
#in parallel
make -C api gen  90.04s user 14.93s system 198% cpu 52.924 total
make -C api gen  91.50s user 15.32s system 198% cpu 53.899 total
make -C api gen  90.09s user 14.93s system 197% cpu 53.251 total

**Review by BupycHuk — APPROVED** [review]

**Review by percona-csalguero — APPROVED** [review]

**Pull request merged**